dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datasetutility/ReplaceFileHandlerIT.java

```diff
@@ -53,6 +101,138 @@ public void shouldCreateDataFile() {
 
     }
 
+    @Test
+    public void shouldReplaceFile() throws IOException {
+        //given
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+
+        Dataverse dataverse = new Dataverse();
+        fillDataverseWithRequiredData(dataverse);
+        em.persist(dataverse);
+
+        Dataset dataset = new Dataset();
+        dataset.setOwner(dataverse);
+        fillDatasetWithExtendedData(dataset);
+        em.persist(dataset);
+        em.flush();
+
+        DataFile initialFile = createTestDataFile(dataset.getLatestVersion(), "banner", "image/png", true);
+        em.persist(initialFile);
+
+
+        byte[] bytes = IOUtils.resourceToByteArray("images/coffeeshop.png", getClass().getClassLoader());
+        File newfile = new File(FileUtil.getFilesTempDirectory() +"/coffeeshop.png");
+        try {
+            Files.write(newfile.toPath(), bytes);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        DataFile newDataFile = createTestDataFile(dataset.getLatestVersion(), "coffeeshop", "image/png", false);
+        newDataFile.setStorageIdentifier("coffeeshop.png");
+        em.persist(newDataFile);
+
+        //when
+        replaceFileHandler.replaceFile(dataset.getFiles().get(0), dataset, newDataFile);
+
+        //then
+        Assert.assertEquals(2 ,dataset.getFiles().size());
+        Assert.assertEquals(1, dataset.getLatestVersion().getFileMetadatas().size());
+        Assert.assertEquals(datafile.findFileMetadataByDatasetVersionId(dataset.getLatestVersion().getId(), 2,"","").size(),
+                dataset.getLatestVersion().getFileMetadatas().size());
+        Assert.assertEquals(datafile.findFileMetadataByDatasetVersionId(dataset.getLatestVersion().getId(), 2,"","").get(0),
+                dataset.getLatestVersion().getFileMetadatas().get(0));
+    }
+    private DataFile createTestDataFile(DatasetVersion datasetVersion, String filename, String fileContentType2, boolean addToDataset) {
```

Comment:
I would split this into two methods:
`createTestDataFile(String filename, String fileContentType)`
and
```
    private void attachDataFileToDataset(DataFile dataFileToAttach, DatasetVersion datasetVersion) {
        FileMetadata fileMetadata = dataFileToAttach.getFileMetadatas().get(0);
        
        dataFileToAttach.setOwner(datasetVersion.getDataset());
        datasetVersion.addFileMetadata(fileMetadata);
        fileMetadata.setDatasetVersion(datasetVersion);
        datasetVersion.getDataset().getFiles().add(dataFileToAttach);
    }
```